spark-commons/src/test/scala/za/co/absa/spark/commons/errorhandling/implementations/ErrorHandlingFilterRowsWithErrorsTest.scala

```diff
@@ -87,4 +89,35 @@ class ErrorHandlingFilterRowsWithErrorsTest extends AnyFunSuite with SparkTestBa
 
     assert(result == expected)
   }
+
+  test("errorColumnType should return a BooleanType") {
+    val errorColumn: ErrorColumn = ErrorHandlingFilterRowsWithErrors.putErrorToColumn(
+      "Test error 1", 1, "This is a test error", Some(errColName))
+
+    val testDf = emptyDf.withColumn(errColName, errorColumn.column)
+    val expectedResults = testDf.col(errColName).expr.dataType
+    val expval = testDf.schema.fields
+
+    val results = ErrorHandlingFilterRowsWithErrors.errorColumnType
+
+    assert(results.defaultSize == expval.length)
+    assert(results == expectedResults)
+  }
+
+  test("errorColumnAggregationType should return None since no column is added during the aggregation") {
+    val errorColumn: ErrorColumn = ErrorHandlingFilterRowsWithErrors.putErrorToColumn(
+      "1st error", 0, "This is an error", Some(errColName)
+    )
+
+    val testDf = emptyDf
+    val expVal = testDf.schema.fields.toList.headOption
```

Comment:
Hmm, I'm not sure if this is relevant here. 

Basically you just have an empty list of columns, basically `None`, but that is still from your input DF, but you are trying to test the datatype after the aggregate operation here. I think I would just remove this. 

But please let me know what you think / your reasoning, maybe I'm just not getting something.

Reply:
You are right, and I spoke with David and when trying to extract the type of the column added to the DF is not doing what is supposed to. so, I robbed it like that meanwhile I'm doing research about it.